### PR TITLE
allow recursive training with a `--recursive` parameter

### PR DESCRIPTION
**Pull request by yoptar**
Tested it with `python -m deeppavlov train --recursive slotfill_dstc2 -d`

### CODE REVIEW

**Comment by yoptar** [comment]
Tested it with `python -m deeppavlov train --recursive slotfill_dstc2 -d`

File: deeppavlov/core/commands/train.py [hunk 1]
```diff
@@ -154,14 +155,20 @@ def get_iterator_from_config(config: dict, data: dict):
...
+                                     to_train=True, to_validate=True, download=False,
+                                     recursive=True) -> Dict[str, Dict[str, float]]:
     """Make training and evaluation of the model described in corresponding configuration file."""
     config = parse_config(config)
 
     if download:
         deep_download(config)
 
+    if to_train and recursive:
+        for subconfig in get_all_elems_from_json(config['chainer'], 'config_path'):
```
Comment (my-master) [inline deeppavlov/core/commands/train.py]: Currently, we can set 'config_path' in any inner attribute of a class in the config.  So I suggest searching 'config_path' recursively in the entire config, not only in first-level elements. Is it possible?
Example from my config (I have to do that to emulate a cycle inside my config):
```
{
        "name": "logit_ranker",
        "squad_model": {
          "config_path": "../deeppavlov/configs/squad/multi_squad_noans.json"
        },
        "top_n": 3,
        "sort_noans": true,
        "in": [
          "chunks",
          "questions"
        ],
        "out": [
          "best_answer",
          "context_indices"
        ]
      },
```

Comment (my-master) [reply deeppavlov/core/commands/train.py]: Oh, I see it's already recursive in `get_all_elems_from_json()`. Alright then!

**Comment by vikmary** [comment]
Checked recursive training for go_bot with trainable slot-filling component.

**Pull request merged**